sklearn/preprocessing/label.py

```diff
@@ -126,12 +126,15 @@ def transform(self, y):
         """
         check_is_fitted(self, 'classes_')
         y = column_or_1d(y, warn=True)
+        # transform of empty array is empty array
+        if _num_samples(y) == 0:
+            return np.array([])
 
         classes = np.unique(y)
         if len(np.intersect1d(classes, self.classes_)) < len(classes):
             diff = np.setdiff1d(classes, self.classes_)
             raise ValueError(
-                    "y contains previously unseen labels: %s" % str(diff))
+                "y contains previously unseen labels: %s" % str(diff))
```

Comment:
Please try to avoid changing unrelated things. It makes your contribution harder to review and may introduce merge conflicts to other pull requests.